code/game/objects/items/devices/scanners.dm

```diff
@@ -258,7 +261,21 @@ REAGENT SCANNER
 				infection_present = 5
 			if(e.has_infected_wound())
 				dat += "\t<span class='scanner'> *Infected wound detected in subject's <b>[limb]</b>. Disinfection recommended.</span>\n"
-
+			if (e.implants.len)
+				for(var/I in e.implants)
+					if(!is_type_in_list(I,known_implants))
+						unknown_body++
+			if(e.hidden)
+				unknown_body++
+			if(e.body_part == UPPER_TORSO) //embryo in chest?
+				if(locate(/obj/item/alien_embryo) in H)
```

Comment:
Show foreign objects if you want, but don't outright detect larvas. Feel free to add this function to the advanced scanner instead.